wsmaster/codenvy-hosted-api-workspace/src/main/java/com/codenvy/api/workspace/server/filters/WorkspacePermissionsFilter.java

```diff
@@ -106,13 +114,26 @@ public void filter(GenericResourceMethod genericResourceMethod, Object[] argumen
             case "updateEnvironment":
             case "addCommand":
             case "deleteCommand":
-            case "updateCommand":
+            case "updateCommand": {
                 workspaceId = ((String)arguments[0]);
                 action = CONFIGURE;
                 break;
+            }
 
-            case "getWorkspaces": //method accessible to every user
+            case "getWorkspaces": {//method accessible to every user
                 return;
+            }
+
+            case "getMachines":
+            case "executeCommandInMachine":
+            case "getProcesses":
+            case "stopProcess":
+            case "getProcessLogs":
+            case "getMachineById": {
+                workspaceId = ((String)arguments[0]);
```

Comment:
I think you can move and adapt these tests https://github.com/codenvy/codenvy/pull/571/files#diff-9d3d8030c3b0a013de5e4c9aa51cc7f2L116